identity/validator.go

```diff
@@ -47,9 +51,13 @@ func (v *Validator) ValidateWithRunner(ctx context.Context, i *Identity, runners
 		return err
 	}
 
+	if len(i.Traits) == 0 {
+		i.Traits = []byte(`{}`)
+	}
```

Comment:
This is the fix. Without, `sjson.SetRawBytes` below would produce invalid JSON, although the payload is valid JSON.